core/db/queries.py

```diff
@@ -1,6 +1,103 @@
+from datetime import datetime
+
+from sqlalchemy import UUID, and_, or_
 from sqlalchemy.orm import joinedload
 
 import core.db.entities as ents
+from core.const import ITLRegion
+from core.util import get_itl_regions_from_postcodes
+
+
+def filter_on_regions(itl_regions: set[str]) -> list[UUID]:
+    """
+    Query DB and return all project UUIDs filtered on region filter params.
+
+    Note: hack due to SQLite legacy issues on local. To be deprecated in light of PostgreSQL array lookups.
+
+    :param itl_regions: List of ITL regions to filter on.
+    :return: List of Project id's filtered on region params.
+    """
+    results = ents.Project.query.with_entities(ents.Project.id, ents.Project.postcodes).distinct().all()
```

Comment:
Documentation for .query() says it is legacy and select() should be used with session.execute() instead. Minor point but could it work in this case?

https://docs.sqlalchemy.org/en/20/orm/queryguide/query.html#sqlalchemy.orm.Query